helix-core/src/register.rs

```diff
@@ -54,15 +46,23 @@ impl Registers {
         self.inner.get(&name)
     }
 
-    pub fn get_mut(&mut self, name: char) -> &mut Register {
-        self.inner
-            .entry(name)
-            .or_insert_with(|| Register::new(name))
+    pub fn get_mut(&mut self, name: char) -> Option<&mut Register> {
+        if name != '_' {
+            Some(
+                self.inner
+                    .entry(name)
+                    .or_insert_with(|| Register::new(name)),
+            )
+        } else {
+            None
+        }
     }
```

Comment:
This method is a bit awkward now because it only returns None for the blackhole register while `Registers::get` returns None for any that doesn't exist. I think we can remove this method and replace it's usage in `helix_term::ui::prompt::Prompt::handle_event` with a new method `Registers::push` that wraps `Register::push`